server/services/nomisPrisonerService.ts

```diff
@@ -4,15 +4,19 @@ import config from '../config'
 import { GetVisitsByFilter, PageVisitIdResponse } from '../@types/nomisPrisoner'
 import logger from '../../logger'
 import { Context } from './nomisMigrationService'
+import type HmppsAuthClient from '../data/hmppsAuthClient'
 
 export default class NomisPrisonerService {
+  constructor(private readonly hmppsAuthClient: HmppsAuthClient) {}
+
   private static restClient(token: string): RestClient {
     return new RestClient('Nomis Prisoner API Client', config.apis.nomisPrisoner, token)
   }
 
-  async getVisitMigrationEstimatedCount(filter: GetVisitsByFilter, context: Context): Promise<PageVisitIdResponse> {
+  async getVisitMigrationEstimatedCount(filter: GetVisitsByFilter, context: Context): Promise<number> {
+    const token = await this.hmppsAuthClient.getSystemClientToken(context.username)
     logger.info(`getting details for visit migration  estimated count`)
```

Comment:
We do need to use system credentials, since this service is protected by system role that can't be assigned to users